openhcl/virt_mshv_vtl/src/processor/mshv/x64.rs

```diff
@@ -933,6 +935,64 @@ impl<'a, 'b> InterceptHandler<'a, 'b> {
     }
 }
 
+impl<'a, 'b> UhX86EmulatorRegisters<'a, 'b, HypervisorBackedX86> for x86emu::CpuState {
+    fn new(vp: &'a mut UhProcessor<'b, HypervisorBackedX86>, vtl: GuestVtl) -> Self {
+        const NAMES: &[HvX64RegisterName] = &[
+            HvX64RegisterName::Rsp,
+            HvX64RegisterName::Es,
+            HvX64RegisterName::Ds,
+            HvX64RegisterName::Fs,
+            HvX64RegisterName::Gs,
+            HvX64RegisterName::Ss,
+            HvX64RegisterName::Cr0,
+            HvX64RegisterName::Efer,
+        ];
+        let mut values = [FromZeroes::new_zeroed(); NAMES.len()];
+        vp.runner
+            .get_vp_registers(vtl, NAMES, &mut values)
+            .expect("register query should not fail");
+
+        let [rsp, es, ds, fs, gs, ss, cr0, efer] = values;
+
+        let mut gps = vp.runner.cpu_context().gps;
+        gps[x86emu::CpuState::RSP] = rsp.as_u64();
```

Comment:
We expect the RSP in the cpu context to be wrong and need updating?

Reply:
Good question

It's not immediately clear to me why the GPs are memory mapped into cpu context, and the other registers are queried. It's also not clear why RSP is present in both the memory map and the query, and we overwrite the value from the memory map. This is not new from this PR though, it's just a port of the existing logic in mshv, can we track and investigate this separately?